app/models/mixins/assignment_mixin.rb

```diff
@@ -81,7 +85,13 @@ def get_assigned_tos
         tag = Tag.find_by(:name => "/" + parts.join("/"))
         result[:tags] << [Classification.find_by(:tag_id => tag.id), klass] unless tag.nil?
       when :label
-        label = CustomAttribute.find_by(:name => parts[1], :value => parts[2])
+        label = if parts[1].starts_with?(BASE64_PREFIX)
+                  name = Base64.decode64(parts[1].slice(BASE64_PREFIX.length + 2..-2))
+                  value = Base64.decode64(parts[2].slice(BASE64_PREFIX.length + 2..-2))
+                  CustomAttribute.find_by(:name => name, :value => value)
+                else
+                  CustomAttribute.find_by(:name => parts[1], :value => parts[2])
```

Comment:
Unrelated but:

- shouldn't CustomAttribute searches here be scoped on `:section => "labels"`?

- parts[1], parts[2] is ugly, consider `managed, name, value = parts`.  Ideally test there are exactly 3 parts (ruby multiple assignment silently assigns nils / ignores extra elements) and first is "managed".
  I wish there was an easy analog to Rails routing for such cases...  There actually is — regexps with named groups, but I'm wary of embracing regexps :-)  Another approach may be dispatching to methods by type, `send("get_assigned_tos_#{parts[0]}", *parts[1..])`.

- why are labels prefixed with `/managed` anyway?